src/v/kafka/server/handlers/describe_configs.cc

```diff
@@ -208,74 +221,101 @@ report_broker_config(describe_configs_result& result, bool include_synonyms) {
         }
     }
 
-    add_broker_config(
-      result,
-      "listeners",
-      config::node().kafka_api,
-      include_synonyms,
-      &kafka_endpoint_format);
+    ss::sstring property_name = "listeners";
+    if (config_property_requested(resource.configuration_keys, property_name)) {
+        add_broker_config(
+          result,
+          property_name,
+          config::node().kafka_api,
+          include_synonyms,
+          &kafka_endpoint_format);
+    }
 
-    add_broker_config(
-      result,
-      "advertised.listeners",
-      config::node().advertised_kafka_api_property(),
-      include_synonyms,
-      &kafka_endpoint_format);
+    property_name = "advertised.listeners";
+    if (config_property_requested(resource.configuration_keys, property_name)) {
+        add_broker_config(
+          result,
+          property_name,
+          config::node().advertised_kafka_api_property(),
+          include_synonyms,
+          &kafka_endpoint_format);
+    }
 
-    add_broker_config(
-      result,
-      "log.segment.bytes",
-      config::shard_local_cfg().log_segment_size,
-      include_synonyms,
-      &describe_as_string<size_t>);
+    property_name = "log.segment.bytes";
+    if (config_property_requested(resource.configuration_keys, property_name)) {
+        add_broker_config(
+          result,
+          property_name,
+          config::shard_local_cfg().log_segment_size,
+          include_synonyms,
+          &describe_as_string<size_t>);
+    }
 
-    add_broker_config(
-      result,
-      "log.retention.bytes",
-      config::shard_local_cfg().retention_bytes,
-      include_synonyms,
-      [](std::optional<size_t> sz) {
-          return ssx::sformat("{}", sz ? sz.value() : -1);
-      });
+    property_name = "log.retention.bytes";
+    if (config_property_requested(resource.configuration_keys, property_name)) {
+        add_broker_config(
+          result,
+          property_name,
+          config::shard_local_cfg().retention_bytes,
+          include_synonyms,
+          [](std::optional<size_t> sz) {
+              return ssx::sformat("{}", sz ? sz.value() : -1);
+          });
+    }
```

Comment:
I wonder if it is possible to merge these operations or create a helper interface such that every instance of `add_broker_config` and `add_config` etc... being called doesn't need to be wrapped in this `config_property_requested` that may make the PR much smaller and easier to understand.

Reply:
I've been thinking about it, but i desided not to change add_*_config function.
If you think, that it better to change this function, I will do it.

Changes would be:
- forward resource.configuration_keys into add_*_config
- change name of function to match functionality (e.g. add_*_config_if_requested) 

There are 3 functions that I would change
add_broker_config,
add_topic_config,
add_config

Reply:
If `add_*_config` were public I'd agree we should think twice about putting too much functionality into them. However, they appear to be defined and used only in describe_configs.cc as well as applying to every instance of add_*_config. 

Would it be as simple as this:?

```c++
static void add_broker_config(... {

if (!config_property_requested(resource.configuration_keys, property_name)) {
  return;
}
```